tests/glsl-intrinsic/shader-subgroup/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,200 @@
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+//#define TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+//#define TEST_when_glsl_subgroupInclusiveAdd_is_not_bugged 
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+#define local_size_x_v 32
+layout(local_size_x = local_size_x_v) in;
+
+__generic<T : __BuiltinLogicalType>
+bool test1Logical() {
+    return true
+#if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        && subgroupAnd(T(1)) == T(1)
+        && subgroupOr(T(1)) == T(1)
+        && subgroupXor(T(1)) == T(0) 
+        && subgroupInclusiveAnd(T(1)) == T(1)
+        && subgroupInclusiveOr(T(1)) == T(1)
+#ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+        && subgroupInclusiveXor(T(1)) == T(0)
+#endif // #ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+        && subgroupExclusiveAnd(T(1)) == T(1)
+        && subgroupExclusiveOr(T(1)) == T(1)
+        && subgroupExclusiveXor(T(1)) == T(1)
+#endif // #if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        ;
+}
+
+__generic<T : __BuiltinLogicalType, let N : int>
+bool testVLogical() {
+    typealias gvec = vector<T, N>;
+    
+    return true
+#if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        && subgroupAnd(gvec(T(1))) == gvec(T(1))
+        && subgroupOr(gvec(T(1))) == gvec(T(1))
+        && subgroupXor(gvec(T(1))) == gvec(T(0))
+        && subgroupInclusiveAnd(gvec(T(1))) == gvec(T(1))
+        && subgroupInclusiveOr(gvec(T(1))) == gvec(T(1))
+#ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_bugged
+        && subgroupInclusiveXor(gvec(T(1))) == gvec(T(0))
+#endif // #ifdef TEST_when_glsl_subgroupInclusiveXor_is_not_buggeds
+        && subgroupExclusiveAnd(gvec(T(1))) == gvec(T(1))
+        && subgroupExclusiveOr(gvec(T(1))) == gvec(T(1))
+        && subgroupExclusiveXor(gvec(T(1))) == gvec(T(1))
+#endif // #if !defined(TARGET_HLSL) && !defined(TARGET_CUDA)
+        ;
+}
+
+bool testLogical() {
+    return true
+        && test1Logical<int>()
+        && testVLogical<int, 2>()
+        && testVLogical<int, 3>()
+        && testVLogical<int, 4>()
+        //&& test1Logical<int8_t>()
+        //&& testVLogical<int8_t, 2>()
+        //&& testVLogical<int8_t, 3>()
+        //&& testVLogical<int8_t, 4>()
+        //&& test1Logical<int16_t>()
+        //&& testVLogical<int16_t, 2>()
+        //&& testVLogical<int16_t, 3>()
+        //&& testVLogical<int16_t, 4>()
+        //&& test1Logical<int64_t>()
+        //&& testVLogical<int64_t, 2>()
+        //&& testVLogical<int64_t, 3>()
+        //&& testVLogical<int64_t, 4>()
```

Comment:
We only need to test int, uint and bool because that's what the spec requires.
        genIType subgroupAnd(genIType value);
        genUType subgroupAnd(genUType value);
        genBType subgroupAnd(genBType value);

Reply:
https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_shader_subgroup_extended_types.txt

I assume from the ext that we will need to support int8 16 and 64 for all subgroup functions eventually (I added the preamble function needed inside main code to handle this, but the code does not work due to a bug with __requireGLSLExtension extension)

Reply:
If we want to keep them, we should make them all working.
Why don't they work when int and uint work?
If the situation is same in HLSL side, we wouldn't make it working for this task, which means we should remove from this test file for the submission because they will be out of the scope for this task.
The task is to bring the parity to HLSL and if HLSL cannot do it, we wouldn't do it for GLSL on this task; we will do it later.

Reply:
it is part of hlsl (should be as per hlsl spec); hlsl.meta handles the hlsl side if it is an int8 (doesn't need special code to handle the case*)

the GLSL side does not work because __requireGLSLExtension does not work with GLSL.meta

Reply:
I am not sure what you mean by "the GLSL side does not work because __requireGLSLExtension does not work with GLSL.meta".
Did you mean that you cannot apply __requireGLSLExtension only to certain types such as int8_t?